crates/rpc-types/src/eth/transaction/receipt.rs

```diff
@@ -68,6 +68,7 @@ impl TransactionReceipt {
             | ReceiptEnvelope::Eip2930(receipt)
             | ReceiptEnvelope::Eip4844(receipt)
             | ReceiptEnvelope::Legacy(receipt) => receipt.receipt.status,
+            _ => false,
```

Comment:
Followup work: 
- `trait Receipt` in alloy-network
- Receipt bound on `type Recept` in `trait Network`
- `fn status` in `trait Receipt`
- `impl Receipt for ReceiptEnvelope` and related types
- change this function to a delegate to `Receipt`